srcalc/src/main/java/gov/va/med/srcalc/db/RiskModelDao.java

```diff
@@ -33,12 +38,30 @@ protected Session getCurrentSession()
      * Returns all RiskModels in the database, in arbitrary order.
      * @return an ImmutableCollection
      */
-    public ImmutableCollection<RiskModel> getAllRiskModels()
+    @SuppressWarnings("unchecked")
+	public ImmutableCollection<RiskModel> getAllRiskModels()
     {
-        @SuppressWarnings("unchecked") // trust Hibernate
-        final List<RiskModel> list =
-                getCurrentSession().createCriteria(RiskModel.class).list();
+        Query q = getCurrentSession().createQuery("from RiskModel");
+        final List<RiskModel> list = q.list();
         return ImmutableList.copyOf(list);
     }
     
+    /**
+     * Returns the RiskModel in the database for the given Id
+     * @return an RiskModel
+     */
+    public RiskModel getRiskModelForId( final int mid ) 
+    {
+        final Query q =  getCurrentSession().createQuery(
```

Comment:
Remember I said Hibernate provides a special-purpose method for this? This whole method body can be replaced with:

    return (RiskModel)getCurrentSession().get(RiskModel.class, mid);